### ISSUE TITLE
Step 3 Ipyrad Running Time Too Long

### ISSUE THREAD

**Issue by Hiers12** [dereneaton/ipyrad#587]
I am running a subsample (2 individuals) of paired end GBS data on Ipyrad, they each have around 12 million reads, and this is a de novo assembly for a plant species in the Asteraceae with no close relatives to use as a reference. When I get to step 3 the clustering seems to sit at 0% no matter how much time and RAM I throw at it, I have run this step at 160GB of memory for 20 hours with no real headway made. I see that it has started writing the clustering files in my directory, but not to completion before slurm ends the job. To troubleshoot this I tightened down the params file and reran step two with the min q score set to 43 (q30) and I hard trimmed the first and last five bp from both forward and reverse reads. This didn't really change the number of reads that passed by very much. I understand that this step can take up to two weeks, but my total sample size is 81 with each having around 12 million reads, so I am concerned if just two samples are taking over 20 hours with this much memory I will be looking at a month long job that takes up a ton of my university's computing resources to run. Are there other bits of the params file I should be looking at making changes to that may help this run a little more efficiently?

**Comment by Hiers12** [comment]
The cluster I am using is undergoing maintenence this week, so I cannot currently access my fastqc results. I am trimming the first and five last bases, and could increase this if need be. It is 150x2 GBS reads, and cluster threshold is set at the default .85. I am curious about the -t function, it looks like it distributes the jobs across cores more evenly? So if I only have 2 samples, each would be given its own core for this step? And if I have 40 cores and 81 samples then each core could handle two samples if I set -t to 40? Or even with 2 samples, if I set -t to the number of cores that I have running it will more efficiently divvy those two samples among those 40?

**Comment by Hiers12** [comment]
@isaacovercast The threading did seem to help a bit, and I increased the RAM to 1.5TB on 40 cores. It ran to about 54% completion of the clustering portion of step 3 in 120 hours. Our IT department has a 120 hour limit on HPC use that they are going to waive for me, but I just want to make sure I have done all I can on my end before starting a 2 week run. Our IT specialist looked at the job today during the clustering and said that it was using all of the cores, but only using 40GB of RAM during the period that he observed it. I see that Ipyrad has uses 75% of the available RAM by default, but 40GB isn't even close to that. Could there be another issue here that I am not seeing?

**Comment by isaacovercast** [comment]
@Hiers12 ipyrad will use as much RAM as it needs of what is available. If it only needs 40GB at a given time then the rest of the RAM won't be touched. Sorry for the delay in response, I hope your run is going well or has completed by this time. Feel free to re-open this issue if something else comes up.
